ProcessLib/HeatTransportBHE/LocalAssemblers/IntegrationPointDataSoil.h

```diff
@@ -14,11 +14,21 @@ namespace ProcessLib
 {
 namespace HeatTransportBHE
 {
-template <typename NodalMatrixType>
+template <typename NodalRowVectorType, typename GlobalDimNodalMatrixType>
 struct IntegrationPointDataSoil final
 {
-    NodalMatrixType const NTN_product_times_w;
-    NodalMatrixType const dNdxTdNdx_product_times_w;
+    IntegrationPointDataSoil(NodalRowVectorType N_,
```

Comment:
I think it is possible to drop the ctor.
@renchao-lu We had this somewhere before. I think here only aggregate initialization is used.

Reply:
At first I tried to drop the ctor and there was a building error sth like "overloaded..cannot take three arguments". Then with the help of Haibing I found that it was because I used emplace_back which conflicted with the struct without ctor. So I replaced with push_back and now it works.